R/decisiontreeutils.R

```diff
@@ -82,30 +81,67 @@ createDTParams <- function(config, data) {
     surrogate <- 2
   }
 
-  rpart_params$usesurrogate <- rxDTree_params$useSurrogate <- surrogate
+  param_list$usesurrogate <- surrogate
 
   # get max bins param
   if(is_XDF && !is.na(as.numeric(config$max.bins))) {
     max_bins <- config$max.bins
     if(max_bins < 2) {
       stop("The minimum bins is 2")
     } else {
-      rxDTree_params$maxNumBins <- max_bins
+      param_list$maxNumBins <- max_bins
     }
   }
 
   # other parameters
-  rpart_params$minsplit <- rxDTree_params$minSplit <- config$min.split
-  rpart_params$minbucket <- rxDTree_params$minBucket <- config$min.bucket
-  rpart_params$xval <- rxDTree_params$xVal <- config$xval.folds
-  rpart_params$maxdepth <- rxDTree_params$maxDepth <- config$max.depth
+  param_list$minsplit <- config$min.split
+  param_list$minbucket <- config$min.bucket
+  param_list$xval <- config$xval.folds
+  param_list$maxdepth <- config$max.depth
 
-  rpart_params$cp <- rxDTree_params$cp <- ifelse(config$the.cp == "Auto" || config$the.cp == "", .00001, config$the.cp)
+  param_list$cp <- ifelse(config$the.cp == "Auto" || config$the.cp == "", .00001, config$the.cp)
```

Comment:
Use `if-else` here. `ifelse` would work, but is generally preferred for vectors of length > 1.